impl/maven-impl/src/main/java/org/apache/maven/impl/AbstractSession.java

```diff
@@ -461,12 +459,12 @@ public org.eclipse.aether.artifact.Artifact toArtifact(ArtifactCoordinates coord
 
     @Override
     public void registerListener(@Nonnull Listener listener) {
-        listeners.add(nonNull(listener));
```

Comment:
this is dedicated in:

- https://github.com/apache/maven/pull/2290 

if possible lets merge this first as this is kind of huge.